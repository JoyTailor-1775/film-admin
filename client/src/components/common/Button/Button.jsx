import React from 'react';
import './Button.scss';

const Button = ({
  text,
  type = 'button',
  color = 'primary',
  size = 'small',
  onClick,
}) => {
  const ButtonActionTypes = Object.freeze({
    button: 'button',
    submit: 'submit',
  });

  const ButtonColorTypes = Object.freeze({
    primary: 'primary',
    error: 'error',
    action: 'action',
    regular: 'regular',
  });

  const ButtonSizeTypes = Object.freeze({
    small: 'small',
    medium: 'medium',
    large: 'large',
  });

  return (
    <button
      type={ButtonActionTypes[type]}
      className={`button ${ButtonColorTypes[color]} ${ButtonSizeTypes[size]}`}
      onClick={onClick}
    >
      {text}
    </button>
  );
};

export default Button;
