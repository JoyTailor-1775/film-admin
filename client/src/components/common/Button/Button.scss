@import '../../../globals/variables.scss';

.button {
  background-color: transparent;
  border: none;
  border-radius: 8px;
  text-transform: uppercase;
  font-weight: 900;
  cursor: pointer;
  padding: 5px 7px;
  box-sizing: border-box;

  &:focus {
    outline: none;
  }
}

.small {
  min-width: 30px;
  height: 30px;
  font-size: 14px;
}

.medium {
  min-width: 35px;
  height: 35px;
  font-size: 16px;
}

.large {
  min-width: 40px;
  height: 40px;
  font-size: 18px;
}

.regular {
  color: #ffffff;
  background-color: $silver;
  border: solid 1px $silver;
  &:hover {
    animation: regularHover 1.5s linear;
  }
}

.primary {
  color: #ffffff;
  background-color: $blue;
  border: solid 1px $blue;
  &:hover {
    animation: primaryHover 1.5s linear;
  }
}

.error {
  color: #ffffff;
  background-color: $red;
  border: solid 1px $red;
  &:hover {
    animation: errorHover 1.5s linear;
  }
}

.action {
  color: #ffffff;
  background-color: $green;
  border: solid 1px $green;
  &:hover {
    animation: actionHover 1.5s linear;
  }
}

@keyframes primaryHover {
  0%,
  100% {
    background-color: $blue;
    color: #ffff;
  }
  50% {
    background-color: #ffff;
    color: $blue;
  }
}

@keyframes errorHover {
  0%,
  100% {
    background-color: $red;
    color: #ffff;
  }
  50% {
    background-color: #ffff;
    color: $red;
  }
}

@keyframes actionHover {
  0%,
  100% {
    background-color: $green;
    color: #ffff;
  }
  50% {
    background-color: #ffff;
    color: $green;
  }
}

@keyframes regularHover {
  0%,
  100% {
    background-color: $silver;
    color: #ffff;
  }
  50% {
    background-color: #ffff;
    color: $silver;
  }
}
