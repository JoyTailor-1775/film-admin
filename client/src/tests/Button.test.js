import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../components/common/Button';

describe('<Button/>', () => {
  test('renders simple button', () => {
    const props = {
      text: 'Test Button',
      onClick: jest.fn(),
      type: 'button',
      color: 'primary',
      size: 'small',
    };

    const component = renderer.create(<Button {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders submit button', () => {
    const props = {
      text: 'Test Button',
      onClick: jest.fn(),
      type: 'submit',
      color: 'primary',
      size: 'small',
    };

    const component = renderer.create(<Button {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders invalid button', () => {
    const props = {
      text: 'Test Button',
      onClick: jest.fn(),
      type: 'sadfd',
      color: 'sdf',
      size: 'dsf',
    };

    const component = renderer.create(<Button {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
