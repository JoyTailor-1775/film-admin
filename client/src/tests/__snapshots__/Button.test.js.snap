// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> renders invalid button 1`] = `
<button
  className="button regular small"
  onClick={[MockFunction]}
>
  Test Button
</button>
`;

exports[`<Button/> renders simple button 1`] = `
<button
  className="button primary small"
  onClick={[MockFunction]}
  type="button"
>
  Test Button
</button>
`;

exports[`<Button/> renders submit button 1`] = `
<button
  className="button primary small"
  onClick={[MockFunction]}
  type="submit"
>
  Test Button
</button>
`;
